samples/shields/lmp90100_evb/rtd/src/main.c

```diff
@@ -60,7 +60,7 @@ static double rtd_temperature(int nom, double resistance)
 
 void main(void)
 {
-	const struct device *lmp90100;
+	const struct device *lmp90100 = DEVICE_DT_GET(DT_INST(0, ti_lmp90100));
```

Comment:
`DEVICE_DT_GET_ANY(ti_lmp90100)` is a better choice for samples

Reply:
`DEVICE_DT_GET(DT_INST(0, ti_lmp90100))` results in a compilation error if no instance is enabled, `DEVICE_DT_GET_ANY(ti_lmp90100)` results in a runtime error.

Reply:
DEVICE_DT_GET_ANY is in fact a better choice for samples. We do not want to use DT_INST in samples. https://github.com/zephyrproject-rtos/zephyr/issues/32139

Reply:
Why? Dropping it will make the sample compile even if misconfigured and only fail at runtime, thus not catching the error in CI (unless ran on a supporting configuration, which I doubt we have in any board cluster).

What is the reason behind your request?

Reply:
It is explained in some detail in the issue I linked.

Reply:
I'm sorry, but I cannot find it in the issue, you linked. I see the argument that some find it confusing, but also ghat you would bring it up on the API meeting before converting samples. 

Has it been brought up yet? I don't agree with designing for runtime errors instead of compile time errors, especially not for samples, which we compile in CI for a reason.

Reply:
> I'm sorry, but I cannot find it in the issue, you linked.

The rationale for removing DT_INST from samples is right there in the issue description.

>  I see the argument that some find it confusing, but also that you would bring it up on the API meeting before converting samples. Has it been brought up yet?

Where did I say I would bring it up in the API meeting? I don't see that mentioned anywhere.

The issue was brought up at dev-review, however, and the consensus was to move forward with the removal of DT_INST from samples as documented in that issue. Here are the minutes:

https://docs.google.com/document/d/15KUINNj7Re0GQiWYpqKvO5jgDw8oiQv3SGJfNHGaPOU/edit#heading=h.1ylyjgx28i5t

(Scroll to 11 February 2021 minutes if that heading anchor doesn't work.)

> I don't agree with designing for runtime errors instead of compile time errors, especially not for samples, which we compile in CI for a reason.

OK, I am sympathetic to that, but that is not an argument for continuing to use DT_INST in a sample.

I guess we could add something like this instead:

```c
#define DEVICE_DT_GET_ANY_OR_COMPILE_ERROR(compat)			    \
	COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(compat),			    \
		    (DEVICE_DT_GET(DT_COMPAT_GET_ANY_STATUS_OKAY(compat))), \
		    (((const struct device*)NULL) +                         \
                     ZERO_OR_COMPILE_ERROR(0)))
```